src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java

```diff
@@ -271,6 +281,31 @@ final void updateChildren(final TaskListener listener) throws IOException, Inter
     }
 
     /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void delete() throws IOException, InterruptedException {
+        checkPermission(DELETE);
+        FolderComputation<I> computation = getComputation();
+        Executor executor = Executor.of(computation);
+        if (executor != null) {
+            LOGGER.log(Level.INFO, "Interrupting {0} in order to delete it", this);
+            executor.interrupt(Result.ABORTED, new CauseOfInterruption.UserInterruption(User.current()));
+            // give it 15 seconds or so to respond to the interrupt
+            long expiration = System.nanoTime() + TimeUnit.SECONDS.toNanos(15);
+            while (executor.isAlive()
+                    && executor.getCurrentExecutable() == computation
+                    && expiration - System.nanoTime() > 0L) {
```

Comment:
System.nanoTime has an arbitrary start that is allowed to be anywhere. So it can wrap at any time. You basically have to do all sums by subtraction to ensure correct handling during roll-over